src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java

```diff
@@ -73,4 +90,156 @@
+     *
+     * @param applicationName is the name of the application to retrieve
+     * @return a {@link CompletableFuture} containing application data such as name, roles, etc.
+     */
+    CompletableFuture<ApplicationOverview> getApplication(String applicationName);
+
+    /**
+     * Request to refresh token of application in Axon Server.
+     *
+     * @param applicationName is the name of the application to refresh token
+     * @return a {@link CompletableFuture} containing new token
+     */
+    CompletableFuture<Token> refreshToken(String applicationName);
+
+    /**
+     * Request to delete application in Axon Server.
+     *
+     * @param applicationName is the name of the application to delete
+     * @return a {@link CompletableFuture} that completes when application has been deleted
+     */
+    CompletableFuture<Void> deleteApplication(String applicationName);
+
+    /**
+     * Request to create Axon Server context.
+     * Returns a {@link CompletableFuture} that completes when the request has been processed by AxonServer.
+     *
+     * @param request the request to create the context, containing the context name, the context roles...
+     * @return a {@link CompletableFuture} that completes when the request has been processed to AxonServer
+     */
+    CompletableFuture<Void> createContext(CreateContextRequest request);
+
+    /**
+     * Request to delete context in Axon Server.
+     *
+     * @param request the request to delete the context, containing the context name and option to preserve event store.
+     * @return a {@link CompletableFuture} that completes when context has been deleted
+     */
+    CompletableFuture<Void> deleteContext(DeleteContextRequest request);
+
+    /**
+     * Request to retrieve context overview in Axon Server.
+     *
+     * @param context is the name of the context to retrieve
+     * @return a {@link CompletableFuture} containing context data such as name, roles, etc.
+     */
+    CompletableFuture<ContextOverview> getContextOverview(String context);
+
+    /**
+     * Request to list all contexts in Axon Server.
+     *
+     * @return a {@link CompletableFuture} that contains list of all contexts in Axon Server.
+     */
+    CompletableFuture<List<ContextOverview>> getAllContexts();
+
+    /**
+     * Subscribes and listens to context updates from Axon Server, like when a context is created or deleted.
+     *
+     * @return a {@link ResultStream} streams context updates.
+     */
+    ResultStream<ContextUpdate> subscribeContextUpdates();
+
+    /**
+     * Request to create replication group.
+     * Returns a {@link CompletableFuture} that completes when the request has been processed by AxonServer.
+     *
+     * @param request the request to create replication group,
+     * containing the replication group name, the node name, role...
+     * @return a {@link CompletableFuture} that completes when the request has been processed to AxonServer
```

Comment:
```suggestion
     * Returns a {@link CompletableFuture} that completes when the request has been processed by Axon Server.
     *
     * @param request the request to create replication group,
     * containing the replication group name, the node name, role...
     * @return a {@link CompletableFuture} that completes when the request has been processed to Axon Server
```

Reply:
I'm unsure about the param description too. Linking the request object would allow users to click through and see what the request contains. That would mean we don't have to name every separate field within the request object.